scipy/sparse/linalg/_svdp.py

```diff
@@ -174,7 +167,8 @@ def _svdp(A, k, which='LM', irl_mode=True, kmax=None,
         ``full_output=True``.
 
     """
-    random_state = check_random_state(random_state)
+    if not isinstance(rng, np.random.RandomState):
+        rng = np.random.default_rng(rng)
```

Comment:
This looks like the code corresponding with:

> Every function call to `_svdp` now specifies rng. It is still an "optional" argument though, so I added code to normalize the input `rng` inside `_svpd` unless it is a `RandomState`. I could put back the `check_random_state` code if that is preferred.

I agree that all calls to `_svdp` specify `rng`, and for the most part they have already normalized the argument (except for a call with `rng=0` in `check_svdp` and `test_examples`). For that reason, what do you think about removing this normalization altogether, since we won't want this special case for `RandomState` to remain after the SPEC 7 transition is complete?